src/org/zaproxy/zap/extension/frontEndScanner/ExtensionFrontEndScanner.java

```diff
@@ -0,0 +1,328 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.frontEndScanner;
+
+import java.awt.CardLayout;
+import java.awt.Font;
+import java.lang.Exception;
+import java.lang.String;
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+import java.util.UUID;
+
+import javax.swing.ImageIcon;
+import javax.swing.JTextPane;
+
+import net.htmlparser.jericho.Element;
+import net.htmlparser.jericho.OutputDocument;
+import net.htmlparser.jericho.Source;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.proxy.ProxyListener;
+import org.parosproxy.paros.extension.AbstractPanel;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.extension.history.ProxyListenerLog;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.extension.api.API;
+import org.zaproxy.zap.utils.FontUtils;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+
+/**
+ * An example ZAP extension which adds a top level menu item, a pop up menu item and a status panel.
+ * <p>
+ * {@link ExtensionAdaptor} classes are the main entry point for adding/loading functionalities provided by the add-ons.
+ *
+ * @see #hook(ExtensionHook)
+ */
+public class ExtensionFrontEndScanner extends ExtensionAdaptor implements ProxyListener {
+
+    // The name is public so that other extensions can access it
+    public static final String NAME = "ExtensionFrontEndScanner";
+
+    // The i18n prefix, by default the package name - defined in one place to make it easier
+    // to copy and change this example
+    protected static final String PREFIX = "frontEndScanner";
+
+    private static final String RESOURCE = "/org/zaproxy/zap/extension/frontEndScanner/resources";
+    private static final String FRONT_END_SCANNER = Constant.getZapHome() + "/frontEndScanner/front-end-scanner.js";
+    private static final String SCRIPTS_FOLDER = Constant.getZapHome() + "/scripts/scripts/";
+
+    private static final ImageIcon ICON = new ImageIcon(
+            ExtensionFrontEndScanner.class.getResource( RESOURCE + "/cake.png"));
+
+    private static final String EXAMPLE_FILE = "example/ExampleFile.txt";

```

Comment:
this isnt used is it? All of the example add-on code that isnt used should be removed.